qa/tasks/cephfs/test_volumes.py

```diff
@@ -4345,6 +4337,298 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
+        trashed_sv_files = self.mount_a.get_shell_stdout(
+            f'sudo ls {trashed_sv_path}').strip().split('\n')
+        return len(trashed_sv_files)
+
+    def _assert_trashed_sv_has_num_of_files(self, trashed_sv_path,
+                                            num_of_files):
+        sv_files = self.mount_a.get_shell_stdout(
+            f'sudo ls {trashed_sv_path}').strip().split('\n')
+
+        self.assertEqual(len(sv_files),  num_of_files)
+
+    def test_when_changed_to_true_ongoing_purge_halts(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        set to true, running "ceph fs subvolume rm" will move the subvolume
+        to trash but (asynchronous) purge of the subvolume won't happen till
+        this config option is not explicitly set to False.
+        '''
+        v = self.volname
+        sv = 'sv1'
+        self.config_set('mgr', self.conf_opt, False)
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+
+        sv_path = self._get_sv_path(v, sv)
+        # adding more files for this test since in once few runs it fails due
+        # to race condition
+        self._do_subvolume_io(sv, number_of_files=200)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        # XXX actual test here: test that purging halts
+        self.config_set('mgr', self.conf_opt, True)
+        trashed_sv_path = self._get_trashed_sv_path(sv, sv_path)
+        NUM_OF_FILES = self._get_num_of_files_in_trashed_sv(trashed_sv_path)
+        time.sleep(2)
+        self._assert_trashed_sv_has_num_of_files(trashed_sv_path,
+                                                 NUM_OF_FILES)
+
+    def test_when_changed_to_false_partly_purged_subvol_purges_fully(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        changed to false, the async purging of a subvolume will resume and
+        also finish, causing trash to be empty.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self._do_subvolume_io(sv, number_of_files=100)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self.config_set('mgr', self.conf_opt, True)
+        time.sleep(2)
+
+        # XXX actual test here: test that purging is resumed and finished
+        self.config_set('mgr', self.conf_opt, False)
+        self.wait_for_trashcan_to_be_empty()
+
+    def test_the_default_value(self):
+        '''
+        Test that MGR config option mgr/volumes/pause_purging_trash is false
+        by default.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+
+        o = self.get_ceph_cmd_stdout(f'config get mgr {self.conf_opt}')
+        o = o.lower().strip()
+        self.assertEqual(o, 'false')
+
+    def test_that_subvols_are_purged_by_default(self):
```

Comment:
Isn't https://github.com/ceph/ceph/pull/54396/files#diff-d3a7e3f3f24fff510b4d2a562b2093257b20b3908748c3432d14e460c449186bR4435 an this test performing similar function?

Reply:
Right. Tests are working fine but some optimizations (like this one) can be made. I mentioned it here - https://github.com/ceph/ceph/pull/54396#issuecomment-2383071861.